dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java

```diff
@@ -891,7 +891,13 @@ protected int createComplementDependentCommand(List<Schedule> schedules, Command
 
         dependentCommand.setTaskDependType(TaskDependType.TASK_POST);
         for (DependentProcessDefinition dependentProcessDefinition : dependentProcessDefinitionList) {
-            dependentCommand.setProcessDefinitionCode(dependentProcessDefinition.getProcessDefinitionCode());
+            long processDefinitionCode = dependentProcessDefinition.getProcessDefinitionCode();
+
+            // get dependent downstream version
+            ProcessDefinition processDefinition = processDefinitionMapper.queryByCode(processDefinitionCode);
+            dependentCommand.setProcessDefinitionVersion(processDefinition.getVersion());
```

Comment:
I think it's better query the processDefinitionVersion through `org/apache/dolphinscheduler/dao/mapper/WorkFlowLineageMapper.xml:148`. So that we can  reduce the database query.

Reply:
ok, thanks.